net/inet_pton.c

```diff
@@ -1,6 +1,147 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * net/inet_pton.c
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
 #include <arpa/inet.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
 
-int inet_pton(int af, const char *restrict src, void *restrict dst)
+/* Accept only addresses in dotted-decimal format ddd.ddd.ddd.ddd */
+static int inet_pton4(const char *src, struct in_addr *dst)
 {
-	return 0;
+	in_addr_t addr4 = 0;
+	unsigned long val;
+	int i;
+	char *endptr;
+
+	for (i = 0; i < 4; i++) {
+		if (!isdigit(*src))
+			return 0;
+
+		val = strtoul(src, &endptr, 10);
+		if (endptr == src || (endptr - src) > 3 || val > 255)
```

Comment:
```suggestion
		if (endptr == src || val > 255)
```
If `endptr - src > 3` (min 4 digits)  condition `val > 255` is true.

Reply:
According to [OpenGroup](https://pubs.opengroup.org/onlinepubs/009695399/functions/inet_ntop.html) `inet_pton` function in IPv4 address case (IPv6 is analogous) shall accept addresses consisting of 4 numbers, which are 1 to 3 digits. This time our implementation of `strtoul` conforms to the docs and accepts leading-zeros strings. So, without this check addresses like this: `0000000000192.00000000168.0001.000000000000001` would be accepted. I guess, you'd agree, we don't want such addresses to be accepted. I have added a similar check in the IPv6 case.